test/coreneuron/test_spikes.py

```diff
@@ -26,47 +35,49 @@ def test_spikes(use_mpi4py=False):
     h.cvode.cache_efficient(1)
 
     pc = h.ParallelContext()
-  
+
     pc.set_gid2node(pc.id()+1, pc.id())
     myobj = h.NetCon(h.soma(0.5)._ref_v, None, sec=h.soma)
     pc.cell(pc.id()+1, myobj)
 
-
-    # NEURON spikes run
+    # NEURON run
     nrn_spike_t = h.Vector()
     nrn_spike_gids = h.Vector()
     pc.spike_record(-1, nrn_spike_t, nrn_spike_gids)
 
     h.run()
-   
+
     nrn_spike_t = nrn_spike_t.to_python()
     nrn_spike_gids = nrn_spike_gids.to_python()
 
-    # CORENEURON spike_record(-1) / spike_record(gidlist):
+    # CORENEURON run
     from neuron import coreneuron
     coreneuron.enable = True
+    coreneuron.file_mode = file_mode
     coreneuron.verbose = 0
     h.stdinit()
     corenrn_all_spike_t = h.Vector()
     corenrn_all_spike_gids = h.Vector()
-    
-    pc.spike_record( -1 if pc.id() == 0 else (pc.id()),
```

Comment:
By removing this we are no longer testing 2nd case when we iterate over PreSyns and match up vectors.